isa/rv64si/csr.S

```diff
@@ -81,6 +81,22 @@ RVTEST_CODE_BEGIN
   andi a0, a0, 1  # a0 = a0 & 1
   beqz a0, finish # if no user mode, skip the rest of these checks
 #endif /* __MACHINE_MODE */
+  la t0, user_mode_end
+  srli t0, t0, 2
+  csrr t1, pmpcfg0
+  csrw pmpaddr0, t0
+  csrr t1, pmpaddr0
+  bne t0, t1, fail
+  li t0, 0x07   # giving read, write and execute permissions
+  or t0, t0, 8  # setting mode to TOR
+  li t1, 255
+  csrrc t1, pmpcfg0, t1
+  csrrs t1, pmpcfg0, t0
+  csrr  t1, pmpcfg0
+  andi  t1, t1, 255
+  bne t0, t1, fail
+  .global no_pmp_implemented
+  no_pmp_implemented:
```

Comment:
Shouldn't this all be guarded by #ifdef __MACHINE_MODE (i.e., should the preceding #endif be moved down here)?  This code shouldn't execute in the S-mode variant of this test.

Reply:
Yes, you are correct